classes/03class/exercises/c03-aws01/starkmatt/c03-aws01.txt

```diff
@@ -0,0 +1,40 @@
+##### Exercise 1 / c01-aws01
+
+- Open EC2 Console
+- Choose Launch Templates, Create launch template.
+- Gave it a name of devopsacademy-launchconfig
+- Select Amazon Linux Image 2, t2.Nano instance, KeyPair (Used Original one created), used devopsacademy-vpc, devops-asg80 Security Group, Select "Assign Public IP address to every instance" (I ran into issues with this before as Only assign a public IP address to default VPC instances was checked)
+the following entry in Advanced / User-Data
+
+#! /bin/bash
+yum update -y
+yum install -y httpd
+curl 169.254.169.254/latest/meta-data/hostname > index.html
+mv index.html /var/www/html/
+systemctl start httpd
+
+- Created a new Security group named "devops-asg80" 
+Inbound Rule:
+Type HTTP / Protocol TCP / Port Range 80 / Source 0.0.0.0/0
+
+
+- Created AutoScaling group using launch template above
+called it devopsa-asg01, with my two public subnets created before public-a / public-b (AZ Southeast A/B)
+Scaling group size 1 / 2 / 2 and balanced with Average CPU Utilization (60%)
+Finished,
+
+Went into Instance and checked on the VM's being created, once green tick's appeared I then navigated to both instances via Web browser
+
+Instance #1: 13.211.47.220
+Instance #2  13.236.84.131
+
+what this line is doing? curl https://169.254.169.254/latest/meta-data/hostname > index.html
+
+Looks like thsi line is calling the metadata of the EC2 Instance and writing it to the index.html.
+The user data then after moves it to /var/www/html and places the file in there so when you navigate to the IP address via the browser it'll
+the curl command results in the html document.
+
+I had some issues with port 80 as the Linix image I used for the build was an older AMI and didn't support systemctl.
+After changing the instance to AMI 2 the build worked with no issues!
```

Comment:
Very well explained! Congrats!

Keep going 😄

Reply:
Thank you Denis 😄 !